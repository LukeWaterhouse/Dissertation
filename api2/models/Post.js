import mongoose from 'mongoose'
var schema = mongoose.Schema({
  userName: { type: String },
  date: { type: String },
  content: { type: String }
})

var Post = mongoose.model('Post', schema)

export default Post
