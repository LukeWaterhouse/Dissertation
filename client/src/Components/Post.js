import React from 'react'

import Box from '@mui/material/Box'
import Card from '@mui/material/Card'
import CardContent from '@mui/material/CardContent'
import Typography from '@mui/material/Typography'

function Post(props) {
  return (
    <div
      data-testid="data-postid"
      style={{ marginRight: '15%', marginLeft: '15%', marginTop: '20px' }}
    >
      <Box sx={{ minWidth: 275 }}>
        <Card variant="outlined">
          <CardContent style={{ backgroundColor: '#2C2F33' }}>
            <Typography
              data-testid="data-postdateid"
              sx={{ fontSize: 14 }}
              color="#97A9B4"
              gutterBottom
            >
              {props.date}
            </Typography>

            <Typography
              data-testid="data-postusernameid"
              sx={{ fontSize: 20 }}
              color="white"
              gutterBottom
            >
              {props.userName}
            </Typography>

            <Typography
              variant="body2"
              color="white"
              data-testid="data-postcontentid"
            >
              {props.content}
            </Typography>
          </CardContent>
        </Card>
      </Box>
    </div>
  )
}

export default Post
