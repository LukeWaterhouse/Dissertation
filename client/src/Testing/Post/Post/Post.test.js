import React from 'react'
import ReactDOM from 'react-dom'
import Post from '../../../Components/Post'
import { render, cleanup, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

afterEach(cleanup)

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Post></Post>, div)
  screen.debug
})

it('renders post correctly', () => {
  const { getByTestId } = render(
    <Post
      date="21/11/25"
      userName="Luke"
      content="content text for testing"
    ></Post>
  )
  expect(getByTestId('data-postdateid')).toHaveTextContent('21/11/25')
  expect(getByTestId('data-postusernameid')).toHaveTextContent('Luke')
  expect(getByTestId('data-postcontentid')).toHaveTextContent(
    'content text for testing'
  )
})

it('post matches snapshot', () => {
  const tree = renderer
    .create(
      <Post
        date="21/11/25"
        userName="Luke"
        content="content text for testing"
      ></Post>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
