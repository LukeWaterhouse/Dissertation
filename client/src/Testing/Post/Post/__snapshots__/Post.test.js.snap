// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post matches snapshot 1`] = `
<div
  data-testid="data-postid"
  style={
    Object {
      "marginLeft": "15%",
      "marginRight": "15%",
      "marginTop": "20px",
    }
  }
>
  <div
    className="MuiBox-root css-12dv9jb"
  >
    <div
      className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        style={
          Object {
            "backgroundColor": "#2C2F33",
          }
        }
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1xx61q6-MuiTypography-root"
          data-testid="data-postdateid"
        >
          21/11/25
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1wd83f6-MuiTypography-root"
          data-testid="data-postusernameid"
        >
          Luke
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-3rq6cm-MuiTypography-root"
          data-testid="data-postcontentid"
        >
          content text for testing
        </p>
      </div>
    </div>
  </div>
</div>
`;
